lib/bls.ex

```diff
@@ -77,4 +77,15 @@ defmodule Bls do
       {:error, _} -> false
     end
   end
+
+  @doc """
+  Converts private to public key
+  """
+  @spec derive_pubkey?(privkey()) :: pubkey()
+  def derive_pubkey?(private_key) do
```

Comment:
The name of this function doesn't match the one on Rust's side. NIFs (modules like these) work by modifying the code inside functions to make them call native functions. For that, you need to have a `derive_pubkey` function on this side too.